conngen/sli/conngen-interface.sli

```diff
@@ -20,66 +20,64 @@
  *
  */
 
-/CGConnect_cg_i_i_l {
-  << >> exch
-  CGConnect_cg_i_i_D_l
-} def
 
-/CGConnect_cg_iV_iV_l {
-  << >> exch
-  CGConnect_cg_iV_iV_D_l
-} def
+/CGConnect trie
 
-/CGConnect_cg_a_a_D_l {
-  4 3 roll  % stack: cg trg param syntype src
-  cv_iv     % convert src to intvectortype
-  4 -3 roll % stack: cg src trg param syntype
-  3 2 roll  % stack: cg trg param syntype src
-  cv_iv     % convert trg to intvectortype
-  3 -2 roll % stack: cg src trg param syntype
-  CGConnect_cg_iV_iV_D_l
-} def
+  % We create the type trie for the list variants using a loop
+  [/arraytype /intvectortype /gidcollectiontype] {
+    /gidlisttype Set
 
-/CGConnect_cg_a_a_l {
-  << >> exch
-  CGConnect_cg_a_a_D_l
-} def
+    % using empty parameter map and default synapse model
+    [/connectiongeneratortype gidlisttype gidlisttype] {
```

Comment:
Yes. This is consistent with the ordinary [connection routines](http://www.nest-simulator.org/connection_management/) in NEST. I think this is acceptable, as `cvgidcollection` should be pretty cheap in most situations. See also https://github.com/nest/nest-simulator/issues/455.